bundles/org.eclipse.equinox.p2.repository/src/org/eclipse/equinox/internal/p2/repository/CacheManager.java

```diff
@@ -82,6 +82,9 @@ public CacheManager(IAgentLocation agentLocation, Transport transport) {
 	private static final String JAR_EXTENSION = ".jar"; //$NON-NLS-1$
 	private static final String XML_EXTENSION = ".xml"; //$NON-NLS-1$
 
+	/** Allows to mute "Using unsafe http transport" warnings */
+	private static final boolean NO_HTTP_WARNING = Boolean.getBoolean("p2.noHttpWarning"); //$NON-NLS-1$
```

Comment:
maybe `p2.skipLocationCheck` would be more generic (just in case we later find to warn about ftp or whatever location type as well)...

Reply:
Good idea, will do tomorrow

Reply:
It seems to me that `location` is so generic a word it doesn't even suggest the location of what.  The installation, the cache, the pool?  Even if it's the location of repository, what about that repository's location is it checking or not?  Something like  *.skipRepositoryProtocolCheck would be more description of what is being checked or not.

I also don't know the extent to which p2's properties are consistently named/prefixed, but others are like this:

	public static final boolean MIRRORS_ENABLED = !"false".equals(Activator.getContext().getProperty("eclipse.p2.mirrors")); //$NON-NLS-1$//$NON-NLS-2$

But a full regex search reveal a number of naming patterns:

![image](https://user-images.githubusercontent.com/208716/184072025-d0dad83c-916d-4578-b93a-bfa93e7e396a.png)

Reply:
Ed, I was looking for `Boolean.getBoolean` / `System.getProperty`matches and that showed me "p2.setSomething" few times, like "p2.pgp.verifyExpiration" or "p2.fragments", but also "eclipse.p2" and "org.eclipse...".

I don't think this is really required to fully qualify "p2" system properties here, doesn't seem to be very important.

But I also see your point about location. 

So I would update it to "p2.skipRepositoryProtocolCheck". That is both specific & generic enough I think.

Reply:
> maybe `p2.skipLocationCheck` would be more generic (just in case we later find to warn about ftp or whatever location type as well)...

Please note that this always requires a started bundle, do not include *any* sanity checks and thus can be a real pain if your code simply fails with "static initialize error null" exceptions. And in most cases simply falls back to system properties if there are no explicit bundle properties.

So I think if one wants to use bundle-properties, then these should actually be assigned inside the activators start method and not in static final fields!

Reply:
> if one wants to use bundle-properties

No. I just want a system property.

Reply:
@iloveeclipse I'm happy with the name you chose.  

As we  noticed, there is not really a consistent naming pattern and I don't think there are many other things name p2 floating around.